numpy/distutils/conv_template.py

```diff
@@ -152,7 +152,7 @@ def parse_values(astr):
 
 
 stripast = re.compile(r"\n\s*\*?")
-named_re = re.compile(r"#\s*(\w*)\s*=([^#]*)#")
+named_re = re.compile(r"#\s*([\S\w]*)\s*=([^#]*)#")
```

Comment:
This changes the meaning of the regex to match all non-space characters, which isn't something we want to do. Can you explain why this change makes any difference?

Reply:
i think you can replace `\s*(\w*)\s*`  with `\s*(\w+\s*)? `

The latter is safe